### PR TITLE
Get history from sql.

### LINKED ISSUE #13585: %paste + history
From a user point of view, it seem like `%paste` should not store itself in IPython's history, but instead store the text that was pasted. It seems unlikely that a user will want to use history to get to paste again over get to the thing that was pasted.

ipython-paste

### PR DESCRIPTION
**Pull request by Carreau**
Fixes #13585

By getting history from sql we can get the transformed history.
This also skip storing history if `%paste` is used and `%paste` itself
will insert the pasted value in history which is more conveninent.

### CODE REVIEW

**Comment by NathanDunfield** [comment]
This PR introduced an incompatibility with `prompt_toolkit 2.0`, see #13645.

**Comment by monester** [comment]
Hello, thanks for this PR - it would definitely work in some cases, but what if I constantly want to copy from tutorial and paste to console. Is there a way to keep %paste in history instead of actual code pasted?

File: IPython/core/interactiveshell.py [hunk 1]
```diff
@@ -3042,9 +3042,8 @@ def error_before_exec(value):
                 cell = raw_cell
```
Comment (Jesuis33) [inline IPython/core/interactiveshell.py:3043]: [license.txt](https://github.com/ipython/ipython/files/8829434/license.txt)

File: IPython/terminal/interactiveshell.py [hunk 2]
```diff
@@ -32,7 +30,7 @@
 from prompt_toolkit.enums import DEFAULT_BUFFER, EditingMode
 from prompt_toolkit.filters import (HasFocus, Condition, IsDone)
 from prompt_toolkit.formatted_text import PygmentsTokens
-from prompt_toolkit.history import InMemoryHistory
+from prompt_toolkit.history import History
```
Comment (Jesuis33) [inline IPython/terminal/interactiveshell.py:33]: fastly compute init --from=https://github.com/fastly/compute-starter-kit-javascript-default

**Pull request merged**